File: FolderSync/iGProgressBarCellManager.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
using TenTec.Windows.iGridLib;
namespace FolderSync
{

	public class iGProgressBarCellManager
	{

		public void Attach(iGrid pGrid, iGCol pCol)
		{
			pCol.CellStyle.CustomDrawFlags = pCol.CellStyle.CustomDrawFlags | iGCustomDrawFlags.Foreground;
			pGrid.CustomDrawCellForeground += CustomDrawCellForeground;
		}

		private void CustomDrawCellForeground(object sender, iGCustomDrawCellEventArgs e)
		{
			iGrid grid = (iGrid)sender;

			object myObjValue = grid.Cells[e.RowIndex, e.ColIndex].Value;

			if (myObjValue == null) {
				return;
			}

			string textDesc = null;
			Rectangle myBounds = e.Bounds;
			myBounds.Inflate(-2, -2);
			myBounds.Width = myBounds.Width - 1;
			myBounds.Height = myBounds.Height - 1;
			if ((myBounds.Width > 0)) {
				e.Graphics.FillRectangle(Brushes.Bisque, myBounds);

				double myValue = 0;
				try {
					myValue = Convert.ToDouble(myObjValue);
					textDesc = "";
				} catch {
					myValue = 0;
					textDesc = myObjValue;
				}

				int myWidth = Convert.ToInt32(myBounds.Width * myValue);
				e.Graphics.FillRectangle(Brushes.SandyBrown, myBounds.X, myBounds.Y, myWidth, myBounds.Height);

				e.Graphics.DrawRectangle(Pens.SaddleBrown, myBounds);

				StringFormat myStringFormat = new StringFormat();
				myStringFormat.Alignment = StringAlignment.Center;
				myStringFormat.LineAlignment = StringAlignment.Center;
				if (string.IsNullOrEmpty(textDesc))
					textDesc = string.Format("{0:F2}%", myValue * 100);
				e.Graphics.DrawString(textDesc, grid.Font, SystemBrushes.ControlText, new Rectangle(myBounds.X, myBounds.Y, myBounds.Width, myBounds.Height), myStringFormat);
			}
		}

	}
}

File: FolderSync/iGProgressBarCellManager.vb
Imports TenTec.Windows.iGridLib

Public Class iGProgressBarCellManager

	Public Sub Attach(ByVal pGrid As iGrid, ByVal pCol As iGCol)
		pCol.CellStyle.CustomDrawFlags = pCol.CellStyle.CustomDrawFlags Or iGCustomDrawFlags.Foreground
		AddHandler pGrid.CustomDrawCellForeground, AddressOf CustomDrawCellForeground
	End Sub

	Private Sub CustomDrawCellForeground(ByVal sender As Object, ByVal e As iGCustomDrawCellEventArgs)
		Dim grid As iGrid = DirectCast(sender, iGrid)

		Dim myObjValue As Object = grid.Cells(e.RowIndex, e.ColIndex).Value

		If myObjValue Is Nothing Then
			Return
    End If

    Dim textDesc As String
		Dim myBounds As Rectangle = e.Bounds
		myBounds.Inflate(-2, -2)
		myBounds.Width = myBounds.Width - 1
		myBounds.Height = myBounds.Height - 1
		If (myBounds.Width > 0) Then
      e.Graphics.FillRectangle(Brushes.Bisque, myBounds)

      Dim myValue As Double
      Try : myValue = CDbl(myObjValue) : textDesc = ""
      Catch : myValue = 0 : textDesc = myObjValue
      End Try

			Dim myWidth As Integer = CInt(myBounds.Width * myValue)
			e.Graphics.FillRectangle(Brushes.SandyBrown, myBounds.X, myBounds.Y, myWidth, myBounds.Height)

			e.Graphics.DrawRectangle(Pens.SaddleBrown, myBounds)

			Dim myStringFormat As StringFormat = New StringFormat
			myStringFormat.Alignment = StringAlignment.Center
      myStringFormat.LineAlignment = StringAlignment.Center
      If textDesc = "" Then textDesc = String.Format("{0:F2}%", myValue * 100)
      e.Graphics.DrawString(textDesc, grid.Font, SystemBrushes.ControlText, New Rectangle(myBounds.X, myBounds.Y, myBounds.Width, myBounds.Height), myStringFormat)
		End If
	End Sub

End Class
